### PR TITLE
OpenAI-DotNet 8.1.2

### PR DESCRIPTION
**Pull request by henduck**
- Added constructor overloads to Tool and Function classes to support manually adding tool calls in the conversation history

### CODE REVIEW

**Comment by StephenHodgson** [comment]
I'll take a look shortly. Thanks for the contribution!

**Comment by jesses-avi** [comment]
Hi @StephenHodgson  ...do you have a sense of when you might be able to look at this PR?  I'm just trying to plan out my own timeline for addressing the use case this covers... and if it will be a long time before this is resolved, I'll just have to pursue an alternative path. I of course understand if this may not be your highest priority project, and appreciate your work on it in any case.

Thanks in advance!!

**Comment by StephenHodgson** [comment]
bump the minor version and I'll run the unit tests and make a release if they pass

**Comment by henduck** [comment]
Awesome... done. Thanks!

File: OpenAI-DotNet/OpenAI-DotNet.csproj [hunk 1]
```diff
@@ -29,8 +29,10 @@ More context [on Roger Pincombe's blog](https://rogerpincombe.com/openai-dotnet-
     <AssemblyOriginatorKeyFile>OpenAI-DotNet.pfx</AssemblyOriginatorKeyFile>
     <IncludeSymbols>true</IncludeSymbols>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <Version>8.1.1</Version>
+    <Version>8.2.0</Version>
```
Comment (StephenHodgson) [inline OpenAI-DotNet/OpenAI-DotNet.csproj:32]: 8.1.2

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
we don't need to bump that far

File: OpenAI-DotNet/OpenAI-DotNet.csproj (hunk 1 above)
Comment (henduck) [reply OpenAI-DotNet/OpenAI-DotNet.csproj]: Gotcha... updated

**Comment by StephenHodgson** [comment]
All unit tests passed locally

**Pull request merged**